### ISSUE TITLE
Digit 9 height inconsistent with rest

### ISSUE THREAD

**Issue by nahsheef** [tonsky/FiraCode#1515]
### Question
Is alignment of digits of default stylistic set (no feature selected) supposed to be equal at top and bottom bounds? I really like Fira Code overall, yet cannot use it at the moment as I fail to ignore the seeming inconsistency. Maybe there's a solution.

### Observation
I cannot quite tell if `9` is misaligned or actually taller than the other digits, but in different terminals on Wayland (foot, alacritty, kitty, wezterm) it sticks out anytime there's a `9` on display within other digits:
![image](https://user-images.githubusercontent.com/128097917/225777314-c5611c9a-d1cd-4223-a9c0-8fd51300d5c3.png)

After taking a screenshot and marking the bottom, it looks like `3` is the same:
![image](https://user-images.githubusercontent.com/128097917/225777503-dad90e27-df2e-43ee-8009-2de762572e77.png)

And looking at the top, `1` doesn't align with `9` and seems like neither does it with `3` (slightly):
![image](https://user-images.githubusercontent.com/128097917/225777934-22aa6162-429c-41d6-967c-e2693c553a9a.png)

Same result with GTK3's font renderer:
![image](https://user-images.githubusercontent.com/128097917/225778521-19537af4-ec0f-4be3-bd55-b54e0177f302.png)

**Comment by kenmcd** [comment]
**Overshoot - Google Fonts Glossary**
https://fonts.google.com/knowledge/glossary/overshoot

**Overshoot (typography) - Wikipedia**
https://en.wikipedia.org/wiki/Overshoot_%28typography%29

**This Optical Illusion Tricks You Into Thinking That Typeface Letters Are the Same Height**
https://slate.com/human-interest/2015/04/tobias-frere-jones-explains-the-optical-illusion-that-tricks-our-brains-into-perceiving-typeface-letters-are-of-equal-height.html

Normal practice.

**Comment by nahsheef** [comment]
Thanks for the typography knowldge, TIL. Is it a deliberate stylistic choice in Fira Code?

**Comment by kenmcd** [comment]
> Is it a deliberate stylistic choice in Fira Code?

Yes. 
You will find overshoot in most fonts.
Overshoot above the cap height, and
overshoot below the baseline.
To appear visually the same height.

**Comment by nahsheef** [comment]
Thanks for sharing more knowledge. Not arguing sytlistic choice, only trying to learn more even if there's no technical solution given the design constraints of the typeface.

> To appear visually the same height.

Does that mean the style is perceived as equal height by humans with a differently wired visual cortex than mine? I mean, `9` and `3`, two round shaped glyphs, distinctly break out of the box for me in source code.

**Comment by tonsky** [comment]
It shouldn’t be that noticeable on low font sizes. Can you show 100% screenshots where you can see 9 and 3 breaking out of the box?

**Comment by nahsheef** [comment]
The screenshots were at 100% (no zoom) yet unrealistic sizes for demonstration purposes, but I get what you're asking. No offense intended.

In Alacritty at size 13 Fira stays inside the box. To check, open the screenshots and zoom in. For what it's worth, I tend to notice details too much for my own good and see 9's overshoot without intentionally focusing on it at sizes 11, 13, 15, and 20 (screenshot omitted).

### size 13
![image](https://user-images.githubusercontent.com/128097917/225994351-61cf3169-452a-4f26-8019-39c756c6a2af.png)

### size 12 miniscule undershoot at bottom, easy to overlook
![image](https://user-images.githubusercontent.com/128097917/225994217-32a6a9e7-5d12-4762-8ffb-1fece9a7f2e0.png)

### size 10 overshoot
![image](https://user-images.githubusercontent.com/128097917/225994434-062c1484-cef8-44b0-b2d1-9ed5e1344b8e.png)

### size 11 overshoot
![image](https://user-images.githubusercontent.com/128097917/225994388-14ee6b1a-c4ab-4410-b4ac-78257b786ed1.png)

### size 15 overshoot
![image](https://user-images.githubusercontent.com/128097917/225994324-dfd4f33b-8026-4bec-a052-13cdcb9f10c0.png)

**Comment by nahsheef** [comment]
If I zoom in and out, I can see a clear undershoot of 9 at some of the small sizes and overshoot at anything above 13. Increasing size further away from 13 seems to render inside the box at some sizes and go back to overshoot at most sizes. Some cyclic back and forth from overshoot to fitting inside the box

**Comment by nahsheef** [comment]
In Alacritty at size 13 Fira stays inside the box. To check, open the screenshots and zoom in.

If I zoom in and out, I can see undershoot at some of the small sizes and overshoot at anything above 13. Increasing size further away from 13 seems to render inside the box at some sizes and go back to overshoot at most sizes. Some cyclic back and forth from overshoot to fitting inside the box.

### size 13
![image](https://user-images.githubusercontent.com/128097917/225994351-61cf3169-452a-4f26-8019-39c756c6a2af.png)

### size 12 miniscule undershoot at bottom, easy to overlook
![image](https://user-images.githubusercontent.com/128097917/225994217-32a6a9e7-5d12-4762-8ffb-1fece9a7f2e0.png)

### size 10 overshoot
![image](https://user-images.githubusercontent.com/128097917/225994434-062c1484-cef8-44b0-b2d1-9ed5e1344b8e.png)

### size 11 overshoot
![image](https://user-images.githubusercontent.com/128097917/225994388-14ee6b1a-c4ab-4410-b4ac-78257b786ed1.png)

### size 15 overshoot
![image](https://user-images.githubusercontent.com/128097917/225994324-dfd4f33b-8026-4bec-a052-13cdcb9f10c0.png)

**Comment by tonsky** [comment]
Wait, you mean at the bottom? Because all your screenshots have perfectly flat top

**Comment by nahsheef** [comment]
Yes, from what I've noticed, it's all at the bottom. 9 jumps from overshoot to fit-in while cycling font size in alacritty. Size 13 is one of the levels where it stays inside. And at size 12 there's a miniscule **under**shoot.
